src/modules/flow/netctl/netctl.c

```diff
@@ -0,0 +1,342 @@
+        -EINVAL);
+    opts = (const struct sol_flow_node_type_netctl_state_options *)
+        options;
+
+    if (opts->name) {
+        mdata->service_name = strdup(opts->name);
+        SOL_NULL_CHECK(mdata->service_name, -ENOMEM);
+    }
+}
+
+static void
+get_state_close(struct sol_flow_node *node, void *data)
+{
+    struct get_state_data *mdata = data;
+
+    free(mdata->service_name);
+}
+
+static int
+get_state(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct get_state_data *mdata = data;
+    const struct sol_ptr_vector *service_list;
+    struct sol_netctl_service *service;
+    uint16_t i = 0;
+    enum sol_netctl_service_state state;
+    int r;
+    const char *name;
+
+    static const char *state_msgs[] = {
+        [SOL_NETCTL_SERVICE_STATE_UNKNOWN] = "Unknown",
+        [SOL_NETCTL_SERVICE_STATE_IDLE] = "Idle",
+        [SOL_NETCTL_SERVICE_STATE_ASSOCIATION] = "Association",
+        [SOL_NETCTL_SERVICE_STATE_CONFIGURATION] = "Configuration",
+        [SOL_NETCTL_SERVICE_STATE_READY] = "Ready",
+        [SOL_NETCTL_SERVICE_STATE_ONLINE] = "Online",
+        [SOL_NETCTL_SERVICE_STATE_DISCONNECT] = "Disconnect",
+        [SOL_NETCTL_SERVICE_STATE_FAILURE] = "Failure",
+        [SOL_NETCTL_SERVICE_STATE_REMOVE] = "Remove",
+    };
+
+    service_list = sol_netctl_get_services();
+
+    while (service = sol_ptr_vector_get(service_list,i)) {
+        name = sol_netctl_service_get_name(service);
+
+        if (name && strcmp(name, mdata->service_name) == 0) {
+            state = sol_netctl_service_get_state(service);
+            r = sol_flow_send_string_packet(node,
+                    SOL_FLOW_NODE_TYPE_NETCTL_STATE__OUT__OUT,
+                    state_msgs[state]);
+            SOL_INT_CHECK(r, < 0, r);
+
+            if (state == SOL_NETCTL_SERVICE_STATE_ONLINE)
+                r = sol_flow_send_bool_packet(node,
+                        SOL_FLOW_NODE_TYPE_NETCTL_STATE__OUT__ONLINE,
+                        true);
+            else
+                r = sol_flow_send_bool_packet(node,
+                        SOL_FLOW_NODE_TYPE_NETCTL_STATE__OUT__ONLINE,
+                        false);
+
+            SOL_INT_CHECK(r, < 0, r);
+
+            return 0;
+        }
+        i++;
+    }
+
+    sol_flow_send_error_packet(node, ENOENT, "Did not found service name %s", mdata->service_name);
+
+    return 0;
+
+}
+
+static int
+enable_radios(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    bool value;
+    int r;
+
+    r = sol_netctl_add_service_monitor(NULL, NULL);
+    SOL_INT_CHECK(r, < 0, r);
+
+    r = sol_flow_packet_get_bool(packet, &value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    SOL_WRN("%d", value);
+    r = sol_netctl_set_radios_offline(value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    value = sol_netctl_get_radios_offline();
```

Comment:
this `enable_radios` must be split into two parts: one for input port, another for output port.

This one will be for input port, rename to `enabled_process` and it will only do `sol_flow_packet_get_bool()` and then `sol_netctl_set_radios_offline(value)`. Without adding a service monitor OR querying the current state. As in another comment I did, you can't immediately query the state, since that value will be updated later, asynchronously, via DBus signal.

The output port will be 2 functions in reality. A node `open` that will call `sol_netctl_add_service_monitor()` with a monitor callback (your second function). From this callback you emit packet on `OUT` port.